lib/saml_idp/request.rb

```diff
@@ -3,6 +3,8 @@
 require 'logger'
 module SamlIdp
   class Request
+    attr_accessor :errors
```

Comment:
Collecting errors for application then application can decide action based on error.